src/appleseed/renderer/modeling/postprocessingstage/colormap.h

```diff
@@ -0,0 +1,73 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Achal Pandey, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.renderer headers
+#include "renderer/modeling/frame/frame.h"
+
+// appleseed.foundation headers
+#include "foundation/image/color.h"
+#include "foundation/image/image.h"
+#include "foundation/math/aabb.h"
+
+// Standard headers.
+#include <vector>
+
+using namespace foundation;
+
+namespace renderer
+{
+
+class ColorMap
+{
+  public:
+    ColorMap();
+
+    void set_palette(std::vector<Color3f> palette);
+
+    void remap_colors(
+      const Frame&    frame,
+      Image*          image,
+      const float     max_val,
+      const float     min_val);
+
+  private:
+    Color3f evaluate_palette(float x);
```

Comment:
Must be const. Also mark `x` as const.

Reply:
Done.

Is marking `x` as const necessary? I modify it in the function's implementation.
`x *= m_palette.size() - 1;`

Reply:
Ah, in that case it's fine!